### PR TITLE
LPS-135136 Create set method for account type

### PR DESCRIPTION
**Pull request by erickmonteiroo**

### CODE REVIEW

**Comment by drewbrokke** [comment]
ci:reevaluate:1297401_179

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountResourceTest.java [hunk 1]
```diff
@@ -268,6 +269,17 @@ public void testPatchOrganizationMoveAccountsByExternalReferenceCode()
...
 
+	@Test(expected = Problem.ProblemException.class)
+	public void testPostAccountInvalidType() throws Exception {
+		Account randomAccount = randomAccount();
+
+		randomAccount.setType(StringUtil.randomString());
+
+		Account postAccount = testPostAccount_addAccount(randomAccount);
+
+		Assert.assertEquals("business", postAccount.getType());
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountResourceTest.java:280]: Should we ever actually reach this line? I imagine we want the `post` call to throw in this case, so if that's true, this line will never be reached. Also, the test _will_ fail if the exception is not thrown, so we'll know something is wrong.

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java [hunk 2]
```diff
@@ -182,7 +182,12 @@ public Account postAccount(Account account) throws Exception {
 		AccountEntry accountEntry = _accountEntryLocalService.addAccountEntry(
 			contextUser.getUserId(), _getParentAccountId(account),
 			account.getName(), account.getDescription(), _getDomains(account),
-			null, null, null, AccountConstants.ACCOUNT_ENTRY_TYPE_BUSINESS,
+			null, null, null,
+			Optional.ofNullable(
+				account.getType()
+			).orElse(
+				AccountConstants.ACCOUNT_ENTRY_TYPE_BUSINESS
+			),
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java:190]: I'd follow the pattern with other properties that might be null and move this into a private method called `_getType`, like `_getParentAccountId` or `_getStatus`.

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountResourceTest.java [hunk 3]
```diff
@@ -330,13 +342,14 @@ public void testPostOrganizationAccountsByExternalReferenceCode()
...
 	protected String[] getAdditionalAssertFieldNames() {
-		return new String[] {"name"};
+		return new String[] {"name", "type"};
 	}
 
 	@Override
 	protected Account randomAccount() throws Exception {
 		Account account = super.randomAccount();
 
+		account.setType(AccountConstants.ACCOUNT_ENTRY_TYPE_BUSINESS);
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/AccountResourceTest.java:352]: SF: sort this alphabetically with the other setters.

**Review by drewbrokke — CHANGES_REQUESTED** [review]
@erickmonteiroo a couple of changes needed still, but we're getting close. 

In addition to the inline comments, we need to also update the `putAccountByExternalReferenceCode` method, since it now acts like an "upsert" operation. That means that if the given ID finds an account, it will update that account, otherwise it will create a new one.

**Comment by erickmonteiroo** [comment]
ci:test:sf

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Review by drewbrokke — APPROVED** [review]

**Comment by drewbrokke** [comment]
ci:forward

**Pull request closed without merging**